Change BASE_URL to CDN servers (#252)

Nebelhom opened the pull request:
This PR address #252 

`BASE_URL` was switched CDN

I tried to create a back_up system in case there are issues with CDN at some point, we automatically fall back to ftp

Nebelhom commented:
@whimboo 

Please review :)

whimboo commented on mozdownload/scraper.py:
```diff
@@ -36,7 +36,10 @@
+BASE_URL = 'http://download.cdn.mozilla.net/pub'
+
+# In case CDN fails
+BACKUP_URL = 'https://ftp.mozilla.org/pub/mozilla.org'
```
I don't think we have to carry the backup URL with us. @nthomas-mozilla what do you say? Is it enough to explicitly work with CDN only? I don't think there will be cases when it will not work.

nthomas-mozilla replied:
AFAIK that CDN URL has everything you need, and it should be reliable.

whimboo replied:
Wonderful. So @Nebelhom please remove the backup logic. We will stay with CDN only, but please make use of HTTPS and not HTTP, which should really be possible.

nthomas-mozilla replied:
I think you'll hit cert issues using https, at least I do loading your BASE_URL in Firefox and getting Akamai. We have an https CDN but it only has installers on it.

whimboo replied:
Ouch, so no way for trustworthy downloads via CDN?

nthomas-mozilla replied:
So https://download-installer.cdn.mozilla.net/pub is everything on ftp except .mar files, and it looks like mozdownload only does 'installers', so just use that domain.

whimboo replied:
We don't only support installer files, but any kind of file which can exist inside a build folder. This included also the packaged tests, the zip build for windows for nightly builds, etc. I had a look at the given URL and was forwarded to ftp.m.o in case for nightly, tinderbox, and try builds. I can see all the files we need even including the mar files. So I'm not sure what you meant that mar files will not be present. Does your comment only apply to candidates and releases? Otherwise it looks good and good to see that there is a way to retrieve files via HTTPS.

Nick what I wonder is if connections are going outside of ftp.m.o to the CDN network for at least the release builds. When I navigate through the folder structure and trying to download various installers, always ftp.mo is used as backend server.

nthomas-mozilla replied:
You weren't actually forwarded to ftp.m.o in the sense of a 302. The headers you get from a request to the CDN are sum of
* the request from CDN to origin server
* your request to the CDN
The origin server is a vhost on the same cluster that serves ftp.m.o, so you get this sort of thing
  X-Backend-Server: ftp5.dmz.scl3.mozilla.com
when making a CDN request, and it doesn't mean you talked to ftp.m.o directly.

The origin server for https://download-installer.cdn.mozilla.net blocks downloads of .mar files at the time of download, but doesn't modify directory listings. eg if you try to download
 https://download-installer.cdn.mozilla.net/pub/firefox/releases/36.0/update/win32/en-US/firefox-34.0.5-36.0.partial.mar
you get a 403.

Nebelhom commented:
@whimboo 

I removed the back_up and changed the `BASE_URL` to https://download-installer.cdn.mozilla.net/pub . That should be it.

Please review :)

whimboo commented:
Given the limitation that we are not able to allow the download of .mar files, we should not switch away from FTP. I'm not sure if we have customers who actually rely on those files, and I don't want to introduce a regression.

Looks like we have to wontfix this feature request, and have to wait until we get something on S3. @nthomas-mozilla do you agree?

whimboo commented:
Sorry Johannes for the wasted time on this PR but I still vote for wontfix. It looks like Nick also doesn't have another option for us. Closing as wontfix.

Nebelhom commented:
@whimboo Don't worry. The aim is to improve mozdownload. If it does not improve it, then what's the point? ;)
